router/src/server.rs

```diff
@@ -1008,6 +1009,10 @@ async fn chat_completions(
     let logprobs = logprobs.unwrap_or(false);
     let tool_prompt = tool_prompt.unwrap_or_default();
     let stop = stop.unwrap_or_default();
+    // enable greedy only when temperature is 0
+    let do_sample = temperature.map_or(true, |t| t > 0.0);
+    let adjusted_temperature = temperature.map_or(1.0, |t| if t == 0.0 { 1.0 } else { t });
+    let temperature = Some(adjusted_temperature);
```

Comment:
Isn´t that.
```rust
let temperature = if temperature == 0.0{
   None
   } else{ Some(temperature) } 
```

Reply:
Better yet, make the condition == 0.0 common to both do_sample and temperature (makes it more readable imho).

Reply:
I don't think that exactly equivalent since we need the following:

1. do_sample to be true in all cases except if the incoming temp is 0
2. we need temperature to be a `Option<f32>` and we need to replace 0 with 1. 

most readable/simple I've come up with so far is...
```rust
let do_sample: bool = temperature.map_or(true, |t| t != 0.0);
let temperature = temperature.map(|t| if t == 0.0 { 1.0 } else { t });
```

whatcha think? happy to make any changes!